rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -1361,6 +1361,37 @@ class DebruijnInterpreter[M[_], F[_]](
         result   <- toSet(baseExpr)
       } yield result
   }
+
+  private[this] val toMap: Method = new Method() {
+    def toMap(baseExpr: Expr): M[Par] =
```

Comment:
Just as `toMap` is only defined on lists when the list elements are tuples, `toMap` should be defined on sets when the set elements are tuples. This is specified in the scala collections library. Please use that as your reference to implement the other collection conversions.